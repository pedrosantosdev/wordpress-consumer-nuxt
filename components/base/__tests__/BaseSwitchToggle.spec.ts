import { describe, it, expect, beforeEach } from 'vitest'
import { render } from '@testing-library/vue'
import BaseSwitchToggleVue from '@/components/base/BaseSwitchToggle.vue'

describe('BaseSwitchToggleVue', () => {
	beforeEach(() => {
		document.body.innerHTML = ''
	})
	it('should render properly without default state', () => {
		const { html } = render(BaseSwitchToggleVue, {
			props: {
				id: 'toggle',
			},
		})
		expect(html()).toMatchSnapshot()
	})
	it('should render properly with default state', () => {
		const { html } = render(BaseSwitchToggleVue, {
			props: {
				id: 'toggle',
				defaultState: true,
			},
		})
		expect(html()).toMatchSnapshot()
	})
	it('should emit toggle when click', async () => {
		const { container, emitted } = render(BaseSwitchToggleVue, {
			props: {
				id: 'toggle',
			},
		})
		const input = container.querySelector('input[type="checkbox"]') as HTMLInputElement
		input.click()
		expect(emitted('toggle')).toBeTruthy()
		expect(emitted('toggle').length).toBe(1)
		expect(emitted('toggle').values().next().value[0]).toBeTruthy()
	})
})
