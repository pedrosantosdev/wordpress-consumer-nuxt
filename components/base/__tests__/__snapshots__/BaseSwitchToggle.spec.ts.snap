// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseSwitchToggleVue > should render properly with default state 1`] = `"<div data-v-f49a3493="" class="relative inline-block w-10 mr-2 align-middle select-none transition duration-200 ease-in"><input data-v-f49a3493="" id="toggle-input" class="toggle-checkbox absolute block w-6 h-6 rounded-full bg-white border-4 appearance-none cursor-pointer" type="checkbox" name="toggle"><label data-v-f49a3493="" for="toggle-input" class="toggle-label block overflow-hidden h-6 rounded-full bg-gray-300 cursor-pointer"></label></div>"`;

exports[`BaseSwitchToggleVue > should render properly without default state 1`] = `"<div data-v-f49a3493="" class="relative inline-block w-10 mr-2 align-middle select-none transition duration-200 ease-in"><input data-v-f49a3493="" id="toggle-input" class="toggle-checkbox absolute block w-6 h-6 rounded-full bg-white border-4 appearance-none cursor-pointer" type="checkbox" name="toggle"><label data-v-f49a3493="" for="toggle-input" class="toggle-label block overflow-hidden h-6 rounded-full bg-gray-300 cursor-pointer"></label></div>"`;
